fastavro/writer.py

```diff
@@ -139,6 +139,67 @@ def prepare_bytes_decimal(data, schema):
     return tmp.getvalue()
 
 
+def prepare_fixed_decimal(data, schema):
+    if not isinstance(data, decimal.Decimal):
+        return data
+    scale = schema['scale']
+    size = schema['size']
+
+    # based on https://github.com/apache/avro/pull/82/
+
+    sign, digits, exp = data.as_tuple()
+
+    if -exp > scale:
+        raise AssertionError(
```

Comment:
Do `AssertionError`s get optimized away in the same way that the `assert` statement does? If so, it might be better to raise a `ValueError` or something that won't be optimized away.

Reply:
FYI, I used the code from https://github.com/apache/avro/pull/82/ as a guide and matched the coding style of the existing ```  prepare_bytes_decimal```, which included this AssertionError. I'm a newbie to the repo and didn't want to change too much. :-)

Anyway, regardless of it "optimizing away", it's probably not the correct error.  The consensus seems to be that assertions are for debugging, and not for bad use of an API.

The avro code uses
```
raise AvroTypeException('Scale provided in schema does not match the decimal')
```

I don't think we need a new error. ```ValueError``` sounds good.

Reply:
Updated, including unit tests.